lib/SILGen/SILGenDecl.cpp

```diff
@@ -1162,8 +1161,35 @@ void SILGenFunction::emitPatternBinding(PatternBindingDecl *PBD,
     SILValue alet;
     {
       SILLocation loc(PBD);
+
+      // Currently we don't pass any options here, so just grab a 'nil'
+      CanType RawPointerTy = C.TheRawPointerType;
+      CanType OptRawPointerTy =
+          OptionalType::get(RawPointerTy)->getCanonicalType();
+//      SILValue options = getOptionalNoneValue(
+//          loc, getTypeLowering(OptRawPointerTy));
```

Comment:
this was my first attempt though this results in a `i64 0` value, which then fails the type check because the function expects an `i8*`.

The function: `declare extern_weak swiftcc void @swift_asyncLet_start(i8* %0, i8* %1, %swift.type* %2, i8* %3, %swift.opaque* %4) #5` (where the %1 is what we need to pass the null too)